x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/index.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ThunkDispatch } from 'redux-thunk';
+import { AnyAction } from 'redux';
+import { connect } from 'react-redux';
+import { MapStoreState } from '../../../reducers/store';
+import { NewVectorLayerEditor } from './wizard';
+import {
+  addLayer,
+  clearDrawingData,
+  setSelectedLayer,
+  setVectorLayerIndexName,
+  updateEditMode,
+  updateFlyout,
+} from '../../../actions';
+import { FLYOUT_STATE } from '../../../reducers/ui';
+import { LayerDescriptor } from '../../../../common/descriptor_types';
+import { getVectorLayerIndexName } from '../../../selectors/map_selectors';
+
+function mapStateToProps(state: MapStoreState) {
```

Comment:
keep `indexName` internal to the actual wizard react-component. Right now, this wizard is an actual connected-component, and its pushing all the temp-state of this component into the global maps redux-store. It probably does not need this overhead.

<---> Other wizards keep their state localized just inside the react-component.  There's nothing really that needs to "persist" in store-state.

So in practice, just keep track of all `indexName` state inside react-component. Until users click the commit- button on the button right, no store-updates need to happen (and that last one will be automatic, with the `RenderWizardArguments#previewLayer` callback.

So I'd model it more after `ClientFileCreateSourceEditor`-wizard, rather than using a connected-component.